### PR TITLE
Feat(all contracts): update to cw-sdk-1.0

### PR DESCRIPTION
**Pull request by maksk7**

### CODE REVIEW

**Comment by maksk7** [comment]
TODO: update astroport-core dependencies to the main branch

**Review by ShadoySV — APPROVED** [review]

File: contracts/generator_controller/src/bps.rs [hunk 1]
```diff
@@ -28,7 +27,8 @@ impl BasicPoints {
 
     pub fn from_ratio(numerator: Uint128, denominator: Uint128) -> Result<Self, ContractError> {
         numerator
-            .checked_multiply_ratio(Self::MAX, denominator)?
+            .checked_multiply_ratio(Self::MAX, denominator)
+            .map_err(|_| StdError::generic_err("Checked multiple ratio error!"))?
```
Comment (epanchee) [inline contracts/generator_controller/src/bps.rs:31]: "multiply ratio"

File: contracts/voting_escrow/src/contract.rs [hunk 2]
```diff
@@ -386,7 +385,7 @@ fn checkpoint(
             if end > point.end && add_amount.is_zero() {
                 // This is extend_lock_time. Recalculating user's voting power
                 let mut lock = LOCKED.load(deps.storage, addr.clone())?;
-                let mut new_voting_power = calc_coefficient(dt).checked_mul(lock.amount)?;
+                let mut new_voting_power = lock.amount * calc_coefficient(dt);
```
Comment (epanchee) [inline contracts/voting_escrow/src/contract.rs:388]: We need to keep checked behaviour. Use [this](https://github.com/astroport-fi/astroport-core/blob/e45685ce20808ddd14e5358a55d9d26128d77744/packages/astroport/src/lib.rs#L47) as an example.

File: contracts/voting_escrow/src/contract.rs [hunk 3]
```diff
@@ -395,7 +394,7 @@ fn checkpoint(
                 slope
             } else {
                 // This is an increase in the user's lock amount
-                let raw_add_voting_power = calc_coefficient(dt).checked_mul(add_amount)?;
+                let raw_add_voting_power = add_amount * calc_coefficient(dt);
```
Comment (epanchee) [inline contracts/voting_escrow/src/contract.rs:397]: Same as commented above

File: packages/astroport-tests/src/lib.rs [hunk 4]
```diff
@@ -11,29 +11,29 @@ pub mod controller_helper;
...
 
-pub fn mock_app() -> TerraApp {
+pub fn mock_app() -> App {
     let mut env = mock_env();
     env.block.time = Timestamp::from_seconds(EPOCH_START);
     let api = MockApi::default();
     let bank = BankKeeper::new();
     let storage = MockStorage::new();
-    let custom = TerraMock::luna_ust_case();
+    //    let custom = TerraMock::luna_ust_case();
```
Comment (epanchee) [inline packages/astroport-tests/src/lib.rs:20]: I think you can remove it completely and the line 27

File: contracts/voting_escrow/src/contract.rs (hunk 2 above)
Comment (maksk7) [reply contracts/voting_escrow/src/contract.rs]: Ok, changed, thanks

File: packages/astroport-tests/src/lib.rs (hunk 4 above)
Comment (maksk7) [reply packages/astroport-tests/src/lib.rs]: Done

File: contracts/voting_escrow/src/contract.rs (hunk 3 above)
Comment (maksk7) [reply contracts/voting_escrow/src/contract.rs]: Ok, done

File: contracts/voting_escrow/src/contract.rs [hunk 5]
```diff
@@ -423,7 +422,7 @@ fn checkpoint(
         let end =
             new_end.ok_or_else(|| StdError::generic_err("Checkpoint initialization error"))?;
         let dt = end - cur_period;
-        add_voting_power = calc_coefficient(dt).checked_mul(add_amount)?;
+        add_voting_power = add_amount * calc_coefficient(dt);
```
[the remark this answers is not in the document]
Comment (maksk7) [reply contracts/voting_escrow/src/contract.rs]: Ok, done

**Review by epanchee — APPROVED** [review]

File: contracts/voting_escrow/Cargo.toml [hunk 6]
```diff
@@ -23,21 +23,21 @@ crate-type = ["cdylib", "rlib"]
...
 [dependencies]
-cw2 = "0.8"
-cw20 = "0.8"
-cw20-base = { version = "0.8", features = ["library"] }
-cosmwasm-std = { version = "0.16" }
-cw-storage-plus = { version = "0.8.0", features = ["iterator"] }
+cw2 = "0.13.3"
+cw20 = "0.13.3"
+cw20-base = { version = "0.13.3", features = ["library"] }
+cosmwasm-std = "1.0.0"
```
Comment (P-Yevhenii) [inline contracts/voting_escrow/Cargo.toml:29]: we discussed and decided which is better to install 1.0

File: contracts/voting_escrow/Cargo.toml [hunk 7]
```diff
@@ -23,21 +23,21 @@ crate-type = ["cdylib", "rlib"]
 backtraces = ["cosmwasm-std/backtraces"]
 
 [dependencies]
-cw2 = "0.8"
-cw20 = "0.8"
-cw20-base = { version = "0.8", features = ["library"] }
-cosmwasm-std = { version = "0.16" }
-cw-storage-plus = { version = "0.8.0", features = ["iterator"] }
+cw2 = "0.13.3"
```
Comment (P-Yevhenii) [inline contracts/voting_escrow/Cargo.toml:26]: version 0.13.4 has already been released. Nowadays, we do not have a binding to the minor version, I recommend to specify 0.13 and for other packages too. To be updated.

**Review by P-Yevhenii — APPROVED** [review]

**Pull request merged**